src/com/visualization/piechart/piechart.js

```diff
@@ -0,0 +1,69 @@
+import {h, Component} 				from 'preact/preact';
+
+import PieSegment					from 'com/visualization/piechart/piesegment';
+
+export default class PieChart extends Component {
+
+	constructor( props ) {
+		super(props);
+
+    }
+
+    // Converts an array of number to an array of percentages
+    convertToPercentage ( values ) {
+        // calculate total
+        let total = 0;
+        values.forEach( ( v ) => ( total += v ) );
+
+        // scale values so there sum is 100%
+        let percentages = values.map( ( v ) => ( (100 * v) / total ) );
+
+        return percentages;
+    }
+
+    render( props ) {
+
+        if ( !(props && props['lables'] && props['values'])) {
+            console.warn('PieChart was created with invalid props', props);
```

Comment:
consider

```
const {labels, values} = props;
if (! (labels && values) )
```

Or even replace props in the 

```
render ({labels, values})
```

if these are the only two fields from props used

Reply:
Actually I'd prefer that we avoid destructuring inside function definitions. By destructuring you lose the props variable, which is useful for a variety of things.

Reply:
Keeping it as `render(props)`